File: leetcode/100-999/125. Valid Palindrome/solution_1.java
class Solution
{
    public String toLowerChars(String s)
    {
        String chars = "";

        for (char ch : s.toLowerCase().toCharArray())
        {
            if (Character.isAlphabetic(ch) || Character.isDigit(ch))
                chars += ch;
        }

        return chars;
    }

    public boolean isPalindrome(String s)
    {
        String chars = toLowerChars(s);
        String reversedChars = new StringBuilder(toLowerChars(s)).reverse().toString();

        return chars.equals(reversedChars);
    }
}

File: leetcode/100-999/125. Valid Palindrome/solution_1.cs
public class Solution
{
    public string toLowerChars(string s)
    {
        string chars = "";

        foreach (var ch in s.ToLower())
        {
            if (char.IsLetter(ch) || char.IsDigit(ch))
                chars += ch;
        }

        return chars;
    }

    public bool IsPalindrome(string s)
    {
        string chars = toLowerChars(s);
        string reversedChars = new string(toLowerChars(s).Reverse().ToArray());

        return chars.Equals(reversedChars);
    }
}